### ISSUE TITLE
PUT docker image end point does not work in vagrant box

### ISSUE THREAD

**Issue by cdeepakroy** [DigitalSlideArchive/HistomicsTK#192]
After provisioning HistomicsTK using vagrant, i tried to add `dsarchive/histomicstk:v0.1.3` through the PUT docker image end point.

This leads to the following error:

![add_docker_image](https://cloud.githubusercontent.com/assets/1923534/18365256/fb41d80c-75df-11e6-8061-bd8b282b06a3.png)

This happens on both osx and ubuntu 14.04

However, when i do this outside the vagrant box on my ubuntu 14.04 machine, i was able to successfully add the docker image and the REST end-points for the CLIs in it were created.

**Comment by cdeepakroy** [comment]
CC: @manthey @jbeezley

**Comment by manthey** [comment]
On the vagrant box, can you ssh in and look to see if there is anything in ~/.girder/logs?  If so, are there any exceptions being logged?  (I added logging to the slicer_cli_web module).

**Comment by cdeepakroy** [comment]
`~/.girder/logs/error.log` is empty. We probably have to log exceptions in all files that are involved in generating these end points for docker image management to pin down what is failing. Surprising it works for you.

@jbeezley Can you please test on your machine and let us know if you see this behavior.

**Comment by manthey** [comment]
I just spun up a new vagrant instance.  My first attempt to PUT a docker image failed.  I restarted the girder service, and it then succeeded.  I've made a PR with some more logging to slicer_cli_web, which should help diagnose the issue.

Two things to try:
- `vagrant ssh` and `sudo service girder restart` and try the PUT again. 
- Navigate to the /opt/histomicstk/girder/plugins/slicer_cli_web directory, checkout the logging branch (which will probably require sudo), and try the PUT again.  Check ~/.girder/logs/error.txt

**Comment by manthey** [comment]
It is possible this worked for me because in my previous test I had shutdown the vagrant box and then restarted it...

**Comment by jbeezley** [comment]
I can confirm that restarting the girder service fixes the issue.

**Comment by manthey** [comment]
The error I see on first run is:
```
[2016-09-09 13:58:08,747] ERROR: Could not find docker image dsarchive/histomicstk:v0.1.3
Traceback (most recent call last):
  File "/opt/histomicstk/girder/plugins/slicer_cli_web/server/models/docker_image_model.py", line 123, in _ImageExistsLocally
    data = self.client.inspect_image(name)
  File "/usr/local/lib/python2.7/dist-packages/docker/utils/decorators.py", line 21, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/docker/api/image.py", line 143, in inspect_image
    self._get(self._url("/images/{0}/json", image)), True
  File "/usr/local/lib/python2.7/dist-packages/docker/utils/decorators.py", line 47, in inner
    return f(self, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 140, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 487, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 585, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 453, in send
    raise ConnectionError(err, request=request)
ConnectionError: ('Connection aborted.', error(13, 'Permission denied'))
Additional info:
  Request URL: PUT http://127.0.0.1:8009/api/v1/HistomicsTK/HistomicsTK/docker_image
  Query string: name=%22dsarchive%2Fhistomicstk%3Av0.1.3%22
  Remote IP: 10.0.2.2
```
I wonder if this is because we install docker as part of the worker role, which happens after the girder role.

**Comment by cdeepakroy** [comment]
Fixed by PR #195.
